File: src/main/java/eu/fabiool/cnpj/file/parsing/lib/model/RecordFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package eu.fabiool.cnpj.file.parsing.lib.model;

import java.text.ParseException;

/**
 *
 * @author Fabio
 */
public class RecordFactory {

    public static InfoCnpj getRecord(final byte[] source) throws ParseException {
        
        switch (source[0]) {
            case '0':
                return Header.parse(source);
            case '1':
                return DadosCadastrais.parse(source);
            case '2':
                return Socio.parse(source);
            case '6':
                return CnaeSecundaria.parse(source);
            case '9':
                return Trailler.parse(source); 
            default:
                break;
        }
        
        throw new IllegalArgumentException("Tipo de Registro Desconhecido");
    }
}

File: src/main/kotlin/model/RecordFactory.kt
package model

import java.text.ParseException

class RecordFactory {

    @Throws(ParseException::class)
    fun getRecord(source: ByteArray): InfoCnpj {

        when (source[0]) {
            48.toByte() -> return Header().parse(source)
            49.toByte() -> return DadosCadastrais().parse(source)
            50.toByte() -> return Socio().parse(source)
            54.toByte() -> return CnaeSecundaria().parse(source)
            57.toByte() -> return Trailler().parse(source)
            else -> {
                throw IllegalArgumentException("Unknown type of registry")
            }

        }
    }
}